### DISCUSSION TITLE
Call For Interest: Apple Music (Darwin only, at first) Segment (and pure-Linux Spotify)

### DISCUSSION THREAD

**Discussion by prokopto-dev** [JanDeDobbeleer/oh-my-posh#1780] [Ideas]
Hey all;

I was poking around and looking at some of the segments; I know I am using this on an OSX system, and am more than excited to be able to track the relevant things in my computing life through my prompt.
I noted we currently don't have an apple music segment. I'm tempted to build one; how much interest is there in seeing this happen?
I also noted there's no default linux (Ubuntu) Spotify build for Spotify segments... Any interest on that as well? Would anyone actually use it?

**Comment by JanDeDobbeleer** [comment]
Sure, if it works. The downside on macOS is probably going to be working with AppleScript again which is slow.

**Comment by prokopto-dev** [comment]
Yeah; it is unfortunate that there's not much CLI-based stuff for accessing
information on OSX-based apps.

I am looking into methods that are a little more robust for the retrieval
of information about apps, both for Windows and OSX, and if possible, Linux
as well.

I saw we currently poll the window title for Spotify, which while pretty
straightforward, definitely makes a lot of assumptions; I also don't know
if the Ubuntu version of Spotify updates the window title at all.

Knowing there's a player API for websites and Apple Music, there has to be
some way that the apps are accessing and interpreting that on Darwin.



On Fri, Feb 18, 2022 at 5:13 PM Jan De Dobbeleer ***@***.***>
wrote:

> Sure, if it works. The downside on macOS is probably going to be working
> with AppleScript again which is slow.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JanDeDobbeleer/oh-my-posh/discussions/1780#discussioncomment-2207809>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHWO6JM2EXAT4G3SEJZHJWDU3277DANCNFSM5OZGHSNA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 


——————————
M./Mr. Courtney C. Caldwell
Ingénieur Logiciel/Application Engineer, Hyannis Port Research
Informatique/Computer Science, S.B. ’20 (UML)
Philosophie/Philosophy, A.B. ’16 (SJC)
GitHub: https://github.com/CapularisPerpetua
